src/System.Configuration.ConfigurationManager/src/System/Configuration/TypeUtil.cs

```diff
@@ -135,6 +141,23 @@ internal static Type GetType(IInternalConfigHost host, string typeString, bool t
             return type;
         }
 
+#if MONO
```

Comment:
No need to push it to CoreFX, make the type partial and keep it in Mono if necessary

Reply:
The #if MONO is confusing here... 

If you were to take corefx sys.config and compare it against refsrc System.Configuration, you would have the following files missing:

ConfigurationBuilder.cs
ConfigurationBuilderChain.cs
ConfigurationBuilderSettings.cs
ConfigurationBuildersSection.cs
ConfigurationStringConstants.cs
Internal/IInternalConfigurationBuilderHost.cs

ConfigurationBuildersSection.cs actually uses these methods...  Which means you could bridge corefx sys.config across the board (mono, .net 4.7, etc).  

That said, I have no problem making it partial and within mono.